src/app/game-state/activity.service.ts

```diff
@@ -154,7 +158,6 @@ export class ActivityService {
           this.characterService.characterState.status.health.value -= 0.01 * this.characterService.characterState.status.health.max;
         }
         // check for activity death
-        this.activityDeath = false;
```

Comment:
This is a required false reduction. Removing it means every death thereafter would be considered an activity death. It's a single activity switch. This has to be included.

Reply:
It's effectively the else statement of a true false if switch.

Reply:
Ah, whoops, didn't notice the variable being used somewhere other than for the achievement unlock.

Reply:
It's still not reset to false for this, since if it's stuck true then future battle deaths etc will give the wrong death message.